### ISSUE TITLE
<intertext> not working inside a webwork exercise

### ISSUE THREAD

**Issue by rkouzes** [rbeezer/mathbook#754]
We have a Webwork exercise set up in our content and part of the solution has an intertext tag that does not properly compile to the html version. Instead it is simply missing as if it weren't there at all. Any idea why this is happening? We use the intertext tag successfully later on the page in a non-Webwork example. Below is the text, with details cut for space.

 <exercise xml:id="example....">
            <webwork>
                <pg-macros>
                    <macro-file>contextLimitedFactor.pl</macro-file>
                </pg-macros>
                <setup>
                    <pg-code>
                        Context("LimitedFactor");
                        Context()->noreduce('(-x)-y','(-x)+y');
                        $answer=Formula("...");
                    </pg-code>
                </setup>
                <statement>
                    <p>...</p>
                </statement>
                <solution>
                    <p>In this exercise, ....
                        <md>
                        <mrow>\amp=4...</mrow>
                        **<intertext>Now examine ....</intertext>**
                        <mrow>\amp=4...</mrow>
                        ...

**Comment by rbeezer** [comment]
Sorry for the delay - I've been traveling, mostly without internet.  I like to reply to bug reports much faster than this.

I think this should mostly be an oversight.  The WW language is a bit restriictive, but we should be able to support this.  @Alex-Jordan knows the details better than me, maybe he can comment on what needs to happen?

**Comment by Alex-Jordan** [comment]
I looked a little closer.

The main issue is that `mathbook-webwork-pg.xsl` has a template for `md`, and that template uses the `mrow`s but does nothing with any `intertext`.

I briefly put some effort into a patch for this. However there is a snag that should be resolved with input from Rob. The snag is that in a WW, `md` doesn't use `align`, `alignat`, or `gather`. It necessarily uses `aligned`, `alignedat`, or `gathered`. So to use any `intertext`s, you have to close one of these `ed` variants. There is a common template to identify if an `md` is using `align`, `alignat`, or `gather`, but not one for `ed` variants. 

So here's an idea for a patch I could implement with Rob's nod of approval. Augment the common template to look upward to see if there is a `webwork` ancestor. And if so, use the `ed` variant. Then modify `md` in `mathbook-webwork-pg.xsl` accordingly to make use of any `intertext`. And also, define `webwork//intertext`.

**Comment by rbeezer** [comment]
Yes, I think modifying the common template to return a "-ed" variant in the case 
of a "webwork" ancestor would  be the cleanest way to go.  Document that it is 
only effective for WW problems (even if the condition seems to make it clear). 
If you use xsl:choose (or it is there already), but more restrictive cases first.

Thanks for investigating this one - sounds pretty much like I expected.
